test/e2e/aks_tags.go

```diff
@@ -182,16 +182,11 @@ func AKSAdditionalTagsSpec(ctx context.Context, inputGetter func() AKSAdditional
 				}
 			}
 
-			Byf("Deleting all tags for machine pool %s", mp.Name)
```

Comment:
Remind me what we were doing before w/ respect to delete all tags via nil data, and why we can't do that anymore?

Reply:
Before, CAPZ had a bit more control with the SDK such that we could delete all tags when `nil` was specified to CAPZ. ASO currently doesn't have a way to distinguish between "I don't care what the tags are" which is how it currently interprets `nil` and "I want the tags to be empty." https://github.com/Azure/azure-service-operator/issues/3367 will enable us to specify `nil` in the ASO spec and it will delete any existing tags and there will no longer be a way to express "I don't care what the tags are" which I don't think we would ever need.

Reply:
Rather than deleting should we comment out and add a // TODO to re-enable when we consume a version of ASO that includes https://github.com/Azure/azure-service-operator/issues/3367 ?

Reply:
The linter doesn't like commented out code, but I'll add a comment.

Reply:
Does this need to be re-added now?

Reply:
Same as https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/4069#discussion_r1391756050